devscripts/bashcompletion.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2017 Endless Mobile, Inc.
+
+import os
+import sys
+
+here = os.path.abspath(os.path.dirname(__file__))
+sys.path.insert(0, os.path.join(here, '..', 'flapjack'))
+
+from flapjack import commands
+
+BASH_COMPLETION_TEMPLATE = os.path.join(here, "bash-completion.in")
+BASH_COMPLETION_FILE = os.path.join(here, '..', 'dist', "flapjack.bash-completion")
+
+def get_command_vars():
+    all_commands = commands._command_registry.keys()
```

Comment:
You could move this inside `commands.py` and thereby avoid accessing private variables from here. (In fact I looked at how flatpak does it, and they have a `flatpak complete` command which doesn't show up in the help, but you can see it for example with `flatpak complete 'flatpak build' 13`.

Reply:
Aha! I looked inside the flatpak code for inspiration and found the hidden `flatpak complete` command, but I couldn't figure the arguments to run it.
I think I'll switch to create a `flapjack complete` hidden command, and move the logic to Python. The bash autocomplete script would then be static and simpler, no need to build it, just to install it.
The only inconvenience is Python infamous startup time, as flatpak will be called each time TAB is pressed to complete, I will keep an eye on that.